esphome/components/rpi_dpi_rgb/display.py

```diff
@@ -0,0 +1,205 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome import pins
+from esphome.components import display
+from esphome.const import (
+    CONF_RESET_PIN,
+    CONF_OUTPUT,
+    CONF_DATA_PINS,
+    CONF_ID,
+    CONF_IGNORE_STRAPPING_WARNING,
+    CONF_DIMENSIONS,
+    CONF_WIDTH,
+    CONF_HEIGHT,
+    CONF_LAMBDA,
+    CONF_COLOR_ORDER,
+    CONF_RED,
+    CONF_GREEN,
+    CONF_BLUE,
+    CONF_NUMBER,
+)
+
+# Temporary work-around
+try:
+    from esphome.const import (
+        CONF_OFFSET_HEIGHT,
+        CONF_OFFSET_WIDTH,
+        CONF_INVERT_COLORS,
+    )
+except ImportError:
+    CONF_OFFSET_HEIGHT = "offset_height"
+    CONF_OFFSET_WIDTH = "offset_width"
+    CONF_INVERT_COLORS = "invert_colors"
+
+CONF_DE_PIN = "de_pin"
+CONF_PCLK_PIN = "pclk_pin"
+CONF_HSYNC_PIN = "hsync_pin"
+CONF_VSYNC_PIN = "vsync_pin"
+
+CONF_HSYNC_FRONT_PORCH = "hsync_front_porch"
+CONF_HSYNC_PULSE_WIDTH = "hsync_pulse_width"
+CONF_HSYNC_BACK_PORCH = "hsync_back_porch"
+CONF_VSYNC_FRONT_PORCH = "vsync_front_porch"
+CONF_VSYNC_PULSE_WIDTH = "vsync_pulse_width"
+CONF_VSYNC_BACK_PORCH = "vsync_back_porch"
+CONF_PCLK_FREQUENCY = "pclk_frequency"
+CONF_PCLK_INVERTED = "pclk_inverted"
+
+rpi_dpi_rgb_ns = cg.esphome_ns.namespace("rpi_dpi_rgb")
+RPI_DPI_RGB = rpi_dpi_rgb_ns.class_("RpiDpiRgb", display.Display, cg.Component)
+ColorOrder = display.display_ns.enum("ColorMode")
+
+COLOR_ORDERS = {
+    "RGB": ColorOrder.COLOR_ORDER_RGB,
+    "BGR": ColorOrder.COLOR_ORDER_BGR,
+}
+DATA_PIN_SCHEMA = pins.gpio_pin_schema(
+    {
+        CONF_OUTPUT: True,
+    },
+    internal=True,
+)
```

Comment:
Why not use the built in types?
```suggestion
DATA_PIN_SCHEMA = pins.internal_gpio_output_pin_schema
```